packages/connectivity/src/scp-cf/destination/get-subscriber-token.ts

```diff
@@ -92,3 +101,63 @@ function getJwtForServiceToken(
     return options.jwt;
   }
 }
+
+/**
+ * @internal
+ * Get a subscriber token pair with required fields. Checks that at least one of the tokens exists and sets defaults if needed.
+ * @returns The decoded subscriber tokens.
+ */
+export function getRequiredSubscriberToken(
+  token: SubscriberToken | undefined
+): Required<SubscriberToken> {
+  if (token) {
+    const { userJwt, serviceJwt } = token;
+
+    if (hasTokens(token)) {
+      // in case one of the tokens is missing use the other token for tenant and user identification
+      // at least one of the properties is defined, therefore ! is safe
+      return {
+        userJwt: userJwt || serviceJwt!,
```

Comment:
our return type is required but with the `!` we do potentially allow the value to be undefined.. is this intended? should we rather throw an error if anything is undefined?

Reply:
oh nevermind, I've only now read the comment..

Reply:
That is what I added the comment for. The problem is, that given the check before it cannot be undefined, but the code to ensure that TS also know that will be a lot more verbose. I agree that this is not a good practice in general, that is why I added the comment. Still, someone could change the code and that makes it potentially dangerous. 
I don't have a strong opinion - I can change the code to be more verbose to make the TS compiler get it too. Let me know what you think.

Reply:
nice, I like that solution 💯